k8s-intf/src/bolero/support.rs

```diff
@@ -0,0 +1,584 @@
+            let addr_mask = u128::MAX.unbounded_shr(mask_len);
+            // /127 addresses are special case where the first and last address are not broadcast or network addresses
+            #[allow(clippy::bool_to_int_with_if)]
+            let smallest_addr = if current_num_prefix_bits == 0 || mask_len >= 127 {
+                0
+            } else {
+                1
+            };
+            #[allow(clippy::bool_to_int_with_if)]
+            let largest_addr = if current_num_prefix_bits == 0 {
+                0 // The address is all prefix, no address bits
+            } else {
+                addr_mask - (if mask_len >= 127 { 0 } else { 1 })
+            };
+            let addr_data = d.gen_u128(
+                Bound::Included(&smallest_addr),
+                Bound::Included(&largest_addr),
+            )?;
+
+            let addr_as_u128 = prefix.unbounded_shl(largest_num_addr_bits) | addr_data;
+            let addr = Ipv6Addr::from(addr_as_u128);
+            addrs.push(format!("{addr}/{mask_len}"));
+            prefix += 1;
+            if prefix > largest_prefix {
+                prefix = 0;
+            }
+        }
+        Some(addrs)
+    }
+}
+
+pub const ALPHA_NUMERIC_CHARS: &str =
+    "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+pub struct SourceMacAddrString(String);
+impl SourceMacAddrString {
+    #[must_use]
+    pub fn inner(&self) -> &str {
+        &self.0
+    }
+}
+
+impl AsRef<str> for SourceMacAddrString {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+// Only generate lower case hex characters for mac addresses
+// because we cannot customize PartialEq for generated types
+// that use this, and we want to be able to compare generated
+// mac addresses with each other without concern for case.
+impl TypeGenerator for SourceMacAddrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        // Generate a random 48-bit MAC address
+        // Set the least significant bit of the mac to 0 for unicast
+        // Start at 2 so the we don't accidentally generate 01:00:00:00:00:00
+        // and then clear it to 00:00:00:00:00:00
+        let mac = d.gen_u64(Bound::Included(&2), Bound::Excluded(&0xffff_ffff_ffff_u64))?
+            & 0xffff_ffff_fffe;
+
+        let bytes = [
+            (mac & 0xff) as u8,
+            ((mac >> 8) & 0xff) as u8,
+            ((mac >> 16) & 0xff) as u8,
+            ((mac >> 24) & 0xff) as u8,
+            ((mac >> 32) & 0xff) as u8,
+            ((mac >> 40) & 0xff) as u8,
+        ];
+        let mac_str = format!(
+            "{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}",
+            bytes[0], bytes[1], bytes[2], bytes[3], bytes[4], bytes[5]
+        );
+        Some(SourceMacAddrString(mac_str))
+    }
+}
+
+pub struct LinuxIfName(pub String);
+const IF_NAME_CHARS: &str = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
+const IF_NAME_MAX_LEN: usize = 16;
```

Comment:
Technically, it is 16 bytes and the last one needs to be null so you really only get 15